xbmc/platform/android/activity/android_main.cpp

```diff
@@ -102,23 +102,25 @@ extern void android_main(struct android_app* state)
 
     CEventLoop eventLoop(state);
     IInputHandler inputHandler;
-    CXBMCApp xbmcApp(state->activity, inputHandler);
-    if (xbmcApp.isValid())
+    CXBMCApp& theApp = CXBMCApp::Create(state->activity, inputHandler);
```

Comment:
It seems odd to me that this `Create` method doesn't return a new object but rather a reference to a new object. Perhaps `Create` returns `void` and you just use the `CXBMCApp::Get()` method to get the reference.

Reply:
"Create returns void" - a factory that returns nothing - feels strange. I'd prefer to keep it as is.